rakam-presto/src/main/java/org/rakam/presto/analysis/PrestoRakamRaptorMetastore.java

```diff
@@ -440,6 +434,48 @@ else if (resultSet.getString(2).equals("total")) {
     }
 
     @Override
+    public List<String> getAttributes(String project, String collection, SchemaField field, Optional<LocalDate> startDate, Optional<LocalDate> endDate, Optional<String> query) {
+
+        if(project == null) {
+            return ImmutableList.of();
+        }
+
+        int samplePercentage = 100;
+        long numRows = 0;
+        String countQuery = format("select sum(shards.row_count) as row_count from tables join shards on (shards.table_id = tables.table_id) where table_name = \"%s\"", collection);
+        try (Handle handle = dbi.open()) {
+            numRows = handle.createQuery(countQuery).map(LongMapper.FIRST).iterator().next();
+        }
+
+        if(numRows > 100000) {
+            samplePercentage = (int) ((100000.0d / numRows) * 100) ;
+        }
+
+        String prestoQuery;
+        prestoQuery = format("select distinct %s from %s.\"%s\".\"%s\" tablesample system (%d)",
+                field.getName(),
+                prestoConfig.getColdStorageConnector(),
+                project,
+                collection,
+                samplePercentage);
+        if(startDate.isPresent() || endDate.isPresent()) {
+            String startDateStr = startDate.isPresent() ? startDate.get().toString() : "1970-1-1";
+            String endDateStr = endDate.isPresent() ? endDate.get().plusDays(1).toString() : LocalDate.now().toString();
+            prestoQuery += format(" where %s BETWEEN date '%s' and date '%s' LIMIT 10",
```

Comment:
If they're not present, you should not add time predicate to the query.

Reply:
It looks like you forgot adding predicate for `field`.

Reply:
If one of them is set, I tried to make it open ended on other side. I can neglect the setted one if the other is not set. Current implementation does not add time predicate if both are not set.

Reply:
added field ( now attribute ) predicate

Reply:
I have limited the time with 30 days max. And also if one is not set the other one automatically sets to 30 days before/after.